crates/red_knot_python_semantic/src/types/diagnostic.rs

```diff
@@ -320,6 +321,36 @@ declare_lint! {
     }
 }
 
+declare_lint! {
+    /// ## What it does
+    /// Checks for invalid arguments to `typing.Annotated`.
+    ///
+    /// ## Why is this bad?
+    /// `Annotated` expects at least two arguments.
+    /// Otherwise, it will raise `TypeError` at runtime.
+    ///
+    /// ## Example
+    ///
+    /// ```python
+    /// from typing import Annotated
+    ///
+    /// x: Annotated[int]
+    /// ```
+    ///
+    /// Use instead:
+    ///
+    /// ```python
+    /// from typing import Annotated
+    ///
+    /// x: Annotated[int, "important metadata"]
+    /// ```
+    pub(crate) static INVALID_ANNOTATED_ARGUMENTS = {
```

Comment:
I'm not sure if we'll really want dedicated separate rules for invalid parameterization of each separate special form? Pyright and mypy both seem to lump these all together, pyright into `reportInvalidTypeForm` and mypy into `valid-type`. That would correspond to our `INVALID_TYPE_FORM`, but we also already have `INVALID_TYPE_PARAMETER` which is more specific. I would be inclined to just use `INVALID_TYPE_PARAMETER` and not create a new rule for this.

I guess we do have the precedent of `INVALID_LITERAL_PARAMETER`, but tbh I'd be inclined to remove that in favor of a more general rule.

@AlexWaygood I haven't reviewed the full discussion history of the PR; is a dedicated rule for this something you asked for in review? What do you think about just using `INVALID_TYPE_PARAMETER` instead?

We can also leave this for a later cleanup/rationalization of rules. (For instance, I like that this correctly uses the term "argument" instead of "parameter", I think we should fix that in other similar rules.)

Reply:
I left review comments on the name of the dedicated rule and the docs for it, but no, I didn't specifically ask for having a dedicated rule. Having a more general rule sounds fine to me. But it should be `INVALID_TYPE_ARGUMENTS` rather than `INVALID_TYPE_PARAMETER`, I think -- the _arguments to the parameters_ are where the invalidity is.

Reply:
Ok, taking that thumbs-up as not-an-objection to the "step further", I'm going to suggest that that is in fact what we should do (roll everything into `INVALID_TYPE_FORM`). I just have a hard time imagining that someone wants to globally suppress "wrong type parameters" as an error class, but not other invalid type forms. And mypy and pyright's behavior suggests there hasn't been demand for this.

@InSyncWithFoo are you up for doing this in this PR? I'd probably prefer that, just so we don't add a new rule and then immediately remove it in another PR.

Reply:
Definitely no objection to ordering rules alphabetically!

Differentiating between diagnostics for things that cause a runtime exception vs things that don't is interesting. In some cases I think that's useful. When it comes to valid vs invalid uses of `typing` module special forms I think it's less useful, because how much "runtime error checking" is done by special forms in `typing` module is very arbitrary and ad-hoc, based mostly on what an implementer felt like doing and was easy to do. Often these behaviors are changed or relaxed if someone can present a good case for wanting to experiment in static analysis with some meaning for some currently-erroring form. So I don't think it's a useful distinction in this case.